src/common/src/typegraph/validator/mod.rs

```diff
@@ -122,9 +125,49 @@ impl<'a> TypeVisitor<'a> for Validator {
     ) -> VisitResult<Self::Return> {
         let type_node = current_node.type_node;
 
+        let tg = context.get_typegraph();
+
+        let get_type_name = |idx: u32| tg.types.get(idx as usize).unwrap().type_name();
+
         if let TypeNode::Function { .. } = type_node {
             // validate materializer??
             // TODO deno static
+        } else if let TypeNode::Either { data, .. } = type_node {
+            let variants = data.one_of.clone();
+            for i in 0..variants.len() {
+                for j in (i + 1)..variants.len() {
+                    let type1 = ExtendedTypeNode::new(tg, variants[i]);
+                    let type2 = ExtendedTypeNode::new(tg, variants[j]);
+
+                    let mut subtype_errors = ErrorCollector::default();
+                    type1.ensure_subtype_of(&type2, tg, &mut subtype_errors);
+
+                    if subtype_errors.errors.is_empty() {
+                        self.push_error(
+                            current_node.path,
+                            format!(
+                                "type '{}' is a subtype of '{}'",
```

Comment:
Can you make it clearer?
Like
`format!("invalid either type: variant #{i} ('{}') is a subtype of variant #{j} ('{}')", ...)`